File: scripts/parse_dr_q.js
const fs = require('fs');
const path = require('path');

const transcriptPath = "C:\\Users\\CARLO\\.gemini\\antigravity-ide\\brain\\e99ec606-e376-48a2-bbac-cdeb8c9bbd94\\.system_generated\\logs\\transcript_full.jsonl";

let userInput = "";
const lines = fs.readFileSync(transcriptPath, 'utf8').split('\n');
for (const line of lines) {
    if (!line) continue;
    try {
        const data = JSON.parse(line);
        if (data.type === "USER_INPUT") {
            if (data.content && data.content.includes("Pregunta 1 – Cardiología")) {
                if (data.content.length > userInput.length) {
                    userInput = data.content; // Take the longest one
                }
            }
        }
    } catch (e) {}
}

if (!userInput) {
    console.error("Could not find user input in transcript.");
    process.exit(1);
}

// Split by "Pregunta " followed by a number
const questionsRaw = userInput.split(/Pregunta \d+\s*[-–]\s*/i).filter(q => q.trim().length > 50);

console.log("Raw user input length: ", userInput.length);
console.log("Found questionsRaw count: ", questionsRaw.length);

const parsedQuestions = [];
for (let i = 0; i < questionsRaw.length; i++) {
    const qText = questionsRaw[i];
    
    // Extract topic
    const qLines = qText.split('\n');
    const headerLine = qLines[0];
    
    let topic = headerLine.trim();
    let difficulty = "hard";
    
    const topicMatch = headerLine.match(/(.*?)\s*\((.*?)\)/);
    if (topicMatch) {
        topic = topicMatch[1].trim();
        const diffRaw = topicMatch[2].trim().toLowerCase();
        if (diffRaw.includes("media")) difficulty = "medium";
        if (diffRaw.includes("baja")) difficulty = "easy";
    }

    // Rest of the text
    const textContent = qLines.slice(1).join('\n');
    
    // Extract statement
    const statementMatch = textContent.match(/([\s\S]*?)\n[A-Z]\./);
    const statement = statementMatch ? statementMatch[1].trim() : textContent.substring(0, 150);
    
    // Extract options
    const options = [];
    const letters = ['A', 'B', 'C', 'D', 'E'];
    for (const letter of letters) {
        const optRegex = new RegExp(`^${letter}\\.\\s*([\\s\\S]*?)(?=\\n[A-E]\\.|\\n\\s*Respuesta correcta|$)`, 'im');
        const optMatch = textContent.match(optRegex);
        if (optMatch) {
            options.push({
                id: letter,
                label: letter,
                text: optMatch[1].trim()
            });
        }
    }
    
    // Extract correct option
    const correctMatch = textContent.match(/Respuesta correcta[:\s]*([A-E])/i);
    const correctOptionId = correctMatch ? correctMatch[1].toUpperCase() : "A";
    
    // Extract "Lo que debes saber para el examen"
    const loQueMatch = textContent.match(/Lo que debes saber para el examen([\s\S]*?)Tema:/i);
    const loQueDebesSaber = loQueMatch ? loQueMatch[1].trim() : "";
    
    // Extract Dr Q explanation
    const drQMatch = textContent.match(/EXPLICACI[OÓ]N DEL PROFE:? BY DR Q([\s\S]*)/i);
    const drQText = drQMatch ? drQMatch[1].trim() : "";
    
    const explanationParts = [];
    if (loQueDebesSaber) {
        explanationParts.push(`**Lo que debes saber para el examen:**\n${loQueDebesSaber}`);
    }
    
    // It's possible drQText was cut, but we'll include whatever we get
    let fullDrQ = `EXPLICACIÓN DEL PROFE BY DR Q\n\n${drQText}`;
    
    // If Dr Q explanation wasn't parsed well, use the rest of the text after correct answer
    if (!drQText && !loQueDebesSaber) {
         const remainderMatch = textContent.match(/Respuesta correcta.*?\n([\s\S]*)/i);
         if (remainderMatch) fullDrQ = remainderMatch[1].trim();
    }
    
    parsedQuestions.push({
        id: `dr-q-eval1-${i+1}`,
        examArea: topic,
        topic: topic,
        difficulty: difficulty,
        tags: ["dr-q", "evaluacion-1", "pro"],
        statement: statement,
        options: options,
        correctOptionId: correctOptionId,
        explanation: explanationParts.join('\n\n'),
        keyPoints: [],
        theoryContent: fullDrQ
    });
}

const tsContent = `import type { TrainingQuestion } from "@/lib/questions/types";\n\nexport const DR_Q_EVAL_1_QUESTIONS: TrainingQuestion[] = ${JSON.stringify(parsedQuestions, null, 2)};\n`;

const outputPath = "d:\\ProyectosDeveloper\\Metodoq\\src\\data\\evaluaciones\\evaluacion-dr-q.ts";
fs.writeFileSync(outputPath, tsContent, "utf8");

console.log(`Successfully extracted ${parsedQuestions.length} questions to ${outputPath}`);

File: scripts/parse_dr_q.py
import json
import re
import os

transcript_path = r"C:\Users\CARLO\.gemini\antigravity-ide\brain\e99ec606-e376-48a2-bbac-cdeb8c9bbd94\.system_generated\logs\transcript_full.jsonl"

user_input = ""
with open(transcript_path, 'r', encoding='utf-8') as f:
    for line in f:
        data = json.loads(line)
        if data.get("type") == "USER_INPUT":
            # Let's check if it contains "Pregunta 1 – Cardiología"
            if "Pregunta 1 – Cardiología" in data.get("content", ""):
                user_input = data["content"]
                break

if not user_input:
    print("Could not find user input in transcript.")
    exit(1)

# Now let's parse it
# Split by "Pregunta " followed by a number
questions_raw = re.split(r'Pregunta \d+\s*[-–]\s*', user_input)
questions_raw = [q for q in questions_raw if q.strip()]

parsed_questions = []
for i, q_text in enumerate(questions_raw):
    # Skip if it doesn't look like a question
    if len(q_text) < 50:
        continue
    
    # Extract topic
    lines = q_text.split('\n')
    header_line = lines[0]
    
    # Topic is usually "Cardiología / Reanimación Cardiopulmonar (alta)"
    topic_match = re.match(r'(.*?)\s*\((.*?)\)', header_line)
    if topic_match:
        topic = topic_match.group(1).strip()
        difficulty_raw = topic_match.group(2).strip()
    else:
        topic = header_line.strip()
        difficulty_raw = "alta"
    
    difficulty = "hard"
    if "media" in difficulty_raw.lower(): difficulty = "medium"
    if "baja" in difficulty_raw.lower(): difficulty = "easy"

    # Rest of the text
    text_content = '\n'.join(lines[1:])
    
    # Extract statement (everything before A.)
    statement_match = re.search(r'(.*?)\nA\.', text_content, re.DOTALL)
    statement = statement_match.group(1).strip() if statement_match else ""
    
    # Extract options
    options_dict = {}
    for letter in ['A', 'B', 'C', 'D', 'E']:
        opt_match = re.search(rf'{letter}\.\s*(.*?)(?=\n[A-E]\.|\n✅|$)', text_content, re.DOTALL)
        if opt_match:
            options_dict[letter] = opt_match.group(1).strip()
            
    # Extract correct option
    correct_match = re.search(r'✅\s*Respuesta correcta[:\s]*([A-E])', text_content)
    correct_option = correct_match.group(1) if correct_match else "A"
    
    # Extract "Lo que debes saber para el examen"
    lo_que_debes_saber_match = re.search(r'Lo que debes saber para el examen(.*?)Tema:', text_content, re.DOTALL)
    lo_que_debes_saber = lo_que_debes_saber_match.group(1).strip() if lo_que_debes_saber_match else ""
    
    # Extract Dr Q explanation
    dr_q_match = re.search(r'EXPLICACIÓN DEL PROFE BY DR Q(.*)', text_content, re.DOTALL)
    dr_q_text = dr_q_match.group(1).strip() if dr_q_match else ""
    
    # Build explanation string
    explanation_parts = []
    if lo_que_debes_saber:
        explanation_parts.append(f"**Lo que debes saber para el examen:**\n{lo_que_debes_saber}")
    
    options = []
    for k, v in options_dict.items():
        options.append({
            "id": k,
            "label": k,
            "text": v
        })
        
    parsed_questions.append({
        "id": f"dr-q-eval1-{i+1}",
        "examArea": topic,
        "topic": topic,
        "difficulty": difficulty,
        "tags": ["dr-q", "evaluacion-1", "pro"],
        "statement": statement,
        "options": options,
        "correctOptionId": correct_option,
        "explanation": "\n\n".join(explanation_parts),
        "theoryContent": f"EXPLICACIÓN DEL PROFE BY DR Q\n\n{dr_q_text}"
    })

# Output to TS file
ts_content = f"""import type {{ TrainingQuestion }} from "@/lib/questions/types";

export const DR_Q_EVAL_1_QUESTIONS: TrainingQuestion[] = {json.dumps(parsed_questions, indent=2, ensure_ascii=False)};
"""

output_path = r"d:\ProyectosDeveloper\Metodoq\src\data\evaluaciones\evaluacion-dr-q.ts"
with open(output_path, "w", encoding="utf-8") as f:
    f.write(ts_content)

print(f"Successfully extracted {len(parsed_questions)} questions to {output_path}")
